src/resty/ljsonschema/init.lua

```diff
@@ -791,16 +796,17 @@ generate_validator = function(ctx, schema)
 
   if schema.minLength or schema.maxLength or schema.pattern or schema.format then
     ctx:stmt(sformat('if %s == "string" then', datatype))
+      ctx:stmt(sformat('  local length = %s(%s) or #%s', ctx:libfunc('custom.str_len'), ctx:param(1), ctx:param(1)))
```

Comment:
Why the `or %s` here?

also indentation seems wrong

Reply:
There is an assumption that the `custom.str_len` function returns the length only when the encoding is the one as expected. With `or %s`, it works as a fallback of `#` operator in case the function returns an error.